packages/kbn-securitysolution-list-utils/src/helpers/index.ts

```diff
@@ -1052,3 +1052,38 @@ export const hasWrongOperatorWithWildcard = (
     }
   });
 };
+
+/**
+ * Event filters helper where given an exceptions list,
+ * determine if both 'subject_name' and 'trusted' are
+ * included in an entry with 'code_signature'
+ */
+export const hasPartialCodeSignatureEntry = (
+  items: ExceptionsBuilderReturnExceptionItem[]
+): boolean => {
+  const os = items[0]?.os_types ?? ['windows'];
+  let name = false;
+  let trusted = false;
+  let nestedName = false;
+  let nestedTrusted = false;
+
+  items[0]?.entries.forEach((e) => {
+    if (e.type === 'nested' && e.field === 'process.Ext.code_signature') {
+      e.entries.forEach((n) => {
+        if (n.field === 'subject_name') {
+          nestedName = true;
+        } else if (n.field === 'trusted') {
+          nestedTrusted = true;
+        }
+      });
+    } else if (
+      e.field === 'process.code_signature.subject_name' ||
+      (os.includes('macos') && e.field === 'process.code_signature.team_id')
+    ) {
+      name = true;
+    } else if (e.field === 'process.code_signature.trusted') {
+      trusted = true;
+    }
+  });
+  return name !== trusted || nestedName !== nestedTrusted;
+};
```

Comment:
I’d recommend simplifying the code to avoid nested iterations. Consider using a single for loop with early returns. Here’s a rough pseudocode example

`
for (e of entries) {
if (e.type === 'nested' && e.field === 'process.Ext.code_signature) {
const includesNestedName = e.entries.some(entry => entry.field === 'subject_name')
const includesNestedTrusted = e.entries.some(entry => entry.field === 'trusted')
if (includesNestedName !== includesNestedTrusted) {
return true // Mismatch found, no need to continue
} else {
e.field === 'process.code_signature.subject_name' ||
      (os.includes('macos') && e.field === 'process.code_signature.team_id')
    ) {
      name = true;
    } else if (e.field === 'process.code_signature.trusted') {
      trusted = true;
    }
}
if (name !== trusted) {
return true; // Mismatch found in outer entries
}
return false; // No mismatch`

Reply:
To avoid `?` and checks for `os` you can go with
`const { os_types = ['windows'], entries = [] } = items[0] || {};` 
if items[0] is undefined then it will default to os_types = ['windows'] and empty entries, otherwise it will be overwritten with what items[0].os_types and .entries carries.

Reply:
I'm wondering if either code block here is going to make sense in 3 months? Can we break it down into smaller chunks of contextual logic? Something like

```json5
isNestedTrustedEntry();
isNestedNameEntry();
isNameEntry();
isTrustedEntry();
```
and use that to compose a boolean statement?

If you do decide to break them into chunks, perhaps it would be a good idea to write unit tests for those so that we understand what that logic does.

Reply:
Good idea @ashokaditya!

Reply:
Hi, I tried to incorporate some of your logic for the single for loop with early returns. I could only do it for the nested condition however because the `name` and `trusted` fields need to be checked in all the entries.